src/models/sourcecodemodel.cpp

```diff
@@ -242,3 +246,39 @@ void SourceCodeModel::setSysroot(const QString& sysroot)
 {
     m_sysroot = sysroot;
 }
+
+void SourceCodeModel::find(const QString& search, Direction direction, int offset)
+{
+    auto searchFunc = [&search](const SourceCodeLine& line) {
+        return line.text.indexOf(search, 0, Qt::CaseInsensitive) != -1;
+    };
+
+    auto start = direction == Direction::Forward
+        ? (m_sourceCodeLines.begin() + offset)
+        : (m_sourceCodeLines.end() - (m_numLines - offset - 1)); // 1 one due to offset of the reverse iterator
+
+    auto it = direction == Direction::Forward
```

Comment:
I believe this code here could be greatly clarified by hoisting it into a function that takes the range as iterators and returns a result index (or -1 if nothing is found).

then you call that with either forward or backward iterators. my hunch is that the code would be much more readable then since you'd only have to take care of the direction when calling the function, but not inside it.

anyhow, leave it as-is for now, I'll try to clean it up myself then